src/main/java/net/dv8tion/jda/internal/JDAImpl.java

```diff
@@ -346,12 +346,10 @@ public void setToken(String token)
 
     public void setStatus(Status status)
     {
-        //noinspection SynchronizeOnNonFinalField
-        synchronized (this.status)
+        synchronized (Status.class) // only for use as a mutex, status is volatile and no counterpart is needed
```

Comment:
I don't think this makes much sense

Reply:
The object being locked originally depended on the current status which is about to be changed, so the original sync block would effectively not have done anything since multiple calls would have acquired locks in different objects. Status.class is consistent and seemed the most appropriate direct substitution. It need not be so as long as it is a consistent object.

Reply:
You should use `synchronized(this)` instead then

Reply:
Seems the only reason the block is needed so StatusChangeEvent isn't potentially getting status == oldStatus, status is volatile in any case
A defensive `if (status != oldStatus) handleEvent` check could work to remove the sync block entirely but would that affect event consumers downstream?

Reply:
The goal is to make sure that status changes aren't skipped when 2 threads try to change it to a different status at once

Reply:
`sync (this)` it is, then